controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -130,16 +132,25 @@ func (r *RegisteredClusterReconciler) Reconcile(computeContextOri context.Contex
 		return ctrl.Result{}, giterrors.WithStack(err)
 	}
 
-	// TODO create managedclusterset for workspace
+	mcsName, err := r.getManagedClusterSetName(ctx, &hubCluster, regCluster)
```

Comment:
It seems like you're really checking whether the managed cluster set exists, and if it doesn't, this mcsName is going to be empty, and it's going to be the wrong value passed down on line 148 since the mcs would have been created in syncManagedClusterSet. Should this function maybe just be getManagedClusterSet? And Perhaps syncManagedClusterSet could call this function, and also it could return the created managed clusterset so oyou can use the name in 148? Something like that... Please give it a ponder and rework it a little.